dl1_data_handler/tests/test_reader.py

```
@@ -35,9 +35,91 @@ def test_r1_waveform_reading(r1_tmp_path, r1_gamma_file):
+        "hillas_width_uncertainty",
+        "hillas_psi",
+        "timing_intercept",
+        "timing_deviation",
+        "timing_slope",
+        "leakage_pixels_width_1",
+        "leakage_pixels_width_2",
+        "leakage_intensity_width_1",
+        "leakage_intensity_width_2",
+        "concentration_cog",
+        "concentration_core",
+        "concentration_pixel",
+        "morphology_n_pixels",
+        "morphology_n_islands",
+        "morphology_n_small_islands",
+        "morphology_n_medium_islands",
+        "morphology_n_large_islands",
+        "intensity_max",
+        "intensity_min",
+        "intensity_mean",
+        "intensity_std",
+        "intensity_skewness",
+        "intensity_kurtosis",
+        "peak_time_max",
+        "peak_time_min",
+        "peak_time_mean",
+        "peak_time_std",
+        "peak_time_skewness",
+        "peak_time_kurtosis",
+        "core_psi",
+    ]
+
+    hillas_names_2 = [
+        "obs_id",
+        "event_id",
+        "tel_id",
+        "hillas_intensity",
+        "hillas_skewness",
+        "hillas_kurtosis",
+        "NO_NAME",  # Include a non-existent name
+    ]
+
+    batch = dl1_image_reader.generate_mono_batch([0])
+
+    # Test with full set of valid names
+    hillas = dl1_image_reader.get_parameters_dict(batch, hillas_names_1)
+    present_count = sum(name in hillas for name in hillas_names_1)
+
+    assert present_count == len(
+        hillas_names_1
+    ), f"Missing parameters: {set(hillas_names_1) - hillas.keys()}"  # nosec
+
+    # Test with one invalid parameter name included
+    hillas_partial = dl1_image_reader.get_parameters_dict(batch, hillas_names_2)
+    present_count_partial = sum(name in hillas_partial for name in hillas_names_2)
+
+    assert present_count_partial < len(
+        hillas_names_2
+    ), "Unexpected match for invalid parameter"  # nosec
+
+    assert (
+        "NO_NAME" not in hillas_partial
+    ), "'NO_NAME' should not be in the result"  # nosec
```

Sorry, just seeing it now. I'd prefer to break if something is requested that is not present in the file rather than silently ignoring the invalid request. I think the minimum should be a warning.

---

warning added

---

Please use the logger from the ctapipe component `self.log.warning(...)`, see [here](https://github.com/cta-observatory/ctapipe/blob/02e63ee9da33b48dee3ad39792befcc12c72ee00/src/ctapipe/core/component.py#L95-L98)